File: Archive/examples/blink.py
#!/usr/bin/python
import time

from gpio_96boards import GPIO

GPIO_A = GPIO.gpio_id('GPIO_A')
pins = (
    (GPIO_A, 'out'),
)


def blink(gpio):
    for i in range(5):
        gpio.digital_write(GPIO_A, GPIO.HIGH)
        time.sleep(i)
        gpio.digital_write(GPIO_A, GPIO.LOW)
        time.sleep(1)


if __name__ == '__main__':
    import argparse

    parser = argparse.ArgumentParser(
        description='Blink LED on GPIO A (pin 23)')
    args = parser.parse_args()

    with GPIO(pins) as gpio:
        blink(gpio)

File: Archive/examples/blink.c
#include <stdio.h>

/* 96BoardsGPIO header file */
#include <gpio.h>

/* Just make the pin numbers easier to remember */

int main(int argc, char * argv[])
{
	int x;
	unsigned int gpio_a = gpio_id("GPIO_A");
	/*
	 * GPIO_A can also be looked up with:
  	 *  gpio_a = gpio_by_letter('A');
	 *  gpio_a = gpio_by_pin(23);
	 */

	// Open the GPIO for use.  Do so by pin number on the
	// Low Speed Expansion Connector.
	if (!gpio_open(gpio_a, "out")) {
		for (x=0; x<10; x++) {
			digitalWrite(gpio_a, HIGH);
			usleep(500000);
			digitalWrite(gpio_a, LOW);
			usleep(500000);
		}
	} else {
		fprintf(stderr, "Unable to open GPIO_A\n");
		return -1;
	}

	return 0;
}
